test/integration/master/audit_dynamic_test.go

```diff
@@ -0,0 +1,263 @@
+	require.NoError(t, err)
+	eventList1.Store(auditinternal.EventList{})
+
+	err = testServerHealthCheck(server2.URL, encoder)
+	require.NoError(t, err)
+	eventList2.Store(auditinternal.EventList{})
+
+	kubeclient, err := kubernetes.NewForConfig(result.ClientConfig)
+	require.NoError(t, err)
+
+	sink1 := auditregv1alpha1.AuditSink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test1",
+		},
+		Spec: auditregv1alpha1.AuditSinkSpec{
+			Policy: auditregv1alpha1.Policy{
+				Level: auditregv1alpha1.LevelRequestResponse,
+				Stages: []auditregv1alpha1.Stage{
+					auditregv1alpha1.StageRequestReceived,
+					auditregv1alpha1.StageResponseStarted,
+					auditregv1alpha1.StageResponseComplete,
+					auditregv1alpha1.StagePanic,
+				},
+			},
+			Webhook: auditregv1alpha1.Webhook{
+				ClientConfig: auditregv1alpha1.WebhookClientConfig{
+					URL: &server1.URL,
+				},
+			},
+		},
+	}
+	sink2 := auditregv1alpha1.AuditSink{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test2",
+		},
+		Spec: auditregv1alpha1.AuditSinkSpec{
+			Policy: auditregv1alpha1.Policy{
+				Level: auditregv1alpha1.LevelRequestResponse,
+				Stages: []auditregv1alpha1.Stage{
+					auditregv1alpha1.StageRequestReceived,
+					auditregv1alpha1.StageResponseStarted,
+					auditregv1alpha1.StageResponseComplete,
+					auditregv1alpha1.StagePanic,
+				},
+			},
+			Webhook: auditregv1alpha1.Webhook{
+				ClientConfig: auditregv1alpha1.WebhookClientConfig{
+					URL: &server2.URL,
+				},
+			},
+		},
+	}
+
+	t.Run("one sink", func(t *testing.T) {
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink1)
+		require.NoError(t, err, "failed to create audit sink1")
+		t.Log("created audit sink1")
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events
+		missing, err := checkForEvents(t, eventList1)
+		require.NoError(t, err, "Failed to match all expected events, events %#v not found!", missing)
+	})
+
+	t.Run("two sink", func(t *testing.T) {
+		eventList1.Store(auditinternal.EventList{})
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink2)
+		require.NoError(t, err, "failed to create audit sink2")
+		t.Log("created audit sink2")
+
+		// perform configmap ops
+		configMapOperations(t, kubeclient)
+
+		// check for corresponding events in both sinks
+		missing, err := checkForEvents(t, eventList1)
+		require.NoError(t, err, "Failed to match all expected events for sink1, events %#v not found!", missing)
+
+		missing, err = checkForEvents(t, eventList2)
+		require.NoError(t, err, "Failed to match all expected events for sink2, events %#v not found!", missing)
+	})
+
+	t.Run("delete sink", func(t *testing.T) {
+		err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Delete("test2", &metav1.DeleteOptions{})
+		require.NoError(t, err, "failed to delete audit sink2")
+		t.Log("deleted audit sink2")
+
+		// sleep to wait for sink to close
+		time.Sleep(5 * time.Second)
```

Comment:
I'm concerned this potentially could flake in the future, depending on how overloaded the CI server is. It's probably ok to go with this for now, but please keep an eye on the *-bazel-verify jobs (which afaik run the integration tests) after this merges in case this starts to flake.